### DISCUSSION TITLE
Add a section

### DISCUSSION THREAD

**Discussion by bluefander** [kdevo/osprey-delight#17] [Q&A] [answered]
Sorry, another question.

I would like to add a section.
What I did:

```
- name: "Work"
      identifier: gallery
      url: "/#work"
      weight: 2
    - name: "Music"
      identifier: music
      url: "/#music"
      weight: 3
    - name: "Blog"
      identifier: blog
      url: "/#blog"
      weight: 4
```

I added a new section called Music in the config.yaml
I made a _music.scss file in assets/sass which is basically a copy of the _about.scss and replaced .about in the beginning of the file with .music.
I wrote "@import "music";" in main.scss
I created a folder "music" in the content folder and created a file music.md with some basic text.

I can see a section called "MUSIC" in the navbar, but I can't see the text I wrote in music.md.

What am I missing?

**Comment by kdevo** [comment, marked as answer]
Hi, 

Adding custom sections is currently not documented (but it's planned, [there is a placeholder for that :smile:](https://github.com/kdevo/osprey-delight/blob/master/CUSTOMIZING.md#build-your-own-sections)). The gist of it is to attach the menu to your content file:

```markdown
---
title: "Music"
menu:
  main:
    identifier: "music"
    url: "/#music"
    weight: 3
---

My music...
``` 

No other adjustments are typically needed. Styling this section can be accomplished via `.default` class or `#music` selector in CSS.

If you want more control over the HTML structure of the section, you can create a partial located at `partials/sections/music.html`. This would be loaded automatically instead of the fallback `partials/sections/default.html` which is used when no `music.html` is found.

In a future release, I'll implement "sideloading" the content by simply placing a `music.md` file in the content directory so that the menu can still be maintained completely independently in `config.yaml`.

**Reply by bluefander** [reply]
Cool, thanks a lot!

**Reply by bluefander** [reply]
Hi kdevo

Recently I have been reworking my website.
I don't quite understand what I am doing wrong and whether I have missed some changes in hugo or the development of your theme, but I have something like this:

```
---
title: "Music"
menu:
  main:
    identifier: "music"
    url: "/#music"
    weight: 2
---
blabla...
```

Now the link that's generated is not "mywebsite.org/#music", but "mywebsite.org/music/music".
So it's copying the structure of my "content" directory.
I am using the dev branch.

Thanks!

**Reply by bluefander** [reply]
Can you in any way reproduce this?

**Reply by kdevo** [reply]
Hi @bluefander 

Can reproduce it. I suspect that this is caused by a recent Hugo change which disregards the `url` within the content file. 
Could you try adding the following to your site `config.yaml` as a workaround:

```yaml
Menu:
  Main:
    - name: "Music"
      identifier: "music" 
      url: "/#music"
      weight: 2
```

Does that help?

**Reply by bluefander** [reply]
That helps at least partially, maybe completely if I know how to apply it properly ;)
So your suggestion actually adds a title called "MUSIC" to the navbar with the url "/#music".
I am not sure whether I should keep these lines in music.md at the same time:
```
---
Title: "Music"
menu:
 main:
   identifier: "music"
   url: "#music"
   weight: 2
---
```

I tried it with them and without them, but on my website the content that's in music.md isn't displayed.

**Reply by kdevo** [reply]
> I am not sure whether I should keep these lines in music.md at the same time

The lines above from the content file can be removed then, since the config.yaml is already configuring it.

> I tried it with them and without them, but on my website the content that's in music.md isn't displayed.

So the section in navbar exists, but the content is missing?

**Reply by bluefander** [reply]
>The lines above from the content file can be removed then, since the config.yaml is already configuring it.

Ok.

> So the section in navbar exists, but the content is missing?

Yes, in my content folder a have a folder called music where there is a file located called music.md. I also tried it when putting this file directly in the content folder, but the content is missing.

I tried various things within the music.md file:

```
(with and without the following three lines)
---
title: "Music"
---

test
test
test
```

Could you write me how the music.md should look like in order to make it work?

**Reply by bluefander** [reply]
In the end it's logical that it doesn't work like this for my music.md file since this
```
Menu:
  Main:
    - name: "Music"
      identifier: "music" 
      url: "/#music"
      weight: 2
```
would refer to a #music.md file in the content folder. I just would like the urls to be consistent on the whole website.

**Reply by bluefander** [reply]
Hi @kdevo

Are there any plans to make this work somewhat easier again in future releases?
Tried with the stable version of the theme and hugo 0.84.0 where this would work, but then I am running into troubles with the icons :sweat_smile:

**Reply by kdevo** [reply]
Yes. The recent Hugo change requires careful re-thinking of the way how the menu is built within the theme.

**Reply by kdevo** [reply]
Since this is likely related to the development version, I'll evaluate/test when I've some more time. Currently there just isn't any space.

**Reply by kdevo** [reply]
Hi @bluefander,

[I've just released v5 of the theme](https://github.com/kdevo/osprey-delight/discussions/34), with many improvements under the hood, a different installation procedure (via hugo modules), and improved documentation. It also fixes above problem and [documents how to properly add a custom section](https://github.com/kdevo/osprey-delight/blob/master/CUSTOMIZING.md#add-your-own-sections).

Side note: This also means that all features from `dev` are now available in the `master` branch.

Hope this helps.
